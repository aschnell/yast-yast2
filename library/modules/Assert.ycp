/**
 * File:	Assert.ycp
 * Summary:	Assertions for test cases
 * Authors:	Martin Vidner <mvidner@suse.cz>
 */
{
module "Assert";

/**
 * @param expected expected value of test
 * @param actual   actual value of test
 * @param fail_message will be logged if test fails
 * @return whether test succeeds
 */
global boolean EqualMsg(any expected, any actual, string fail_message) {
    if (expected == actual) {
        return true;
    }
    else {
        y2error ("%1", fail_message);
        return false;
    }
}   

/**
 * @param expected expected value of test
 * @param actual   actual value of test
 * @return whether test succeeds
 */
global boolean Equal(any expected, any actual) {
    string fail_message = sformat ("assertion failure, expected '%1', got '%2'", expected, actual);
    return EqualMsg(expected, actual, fail_message);
}   

/* EOF */
}
