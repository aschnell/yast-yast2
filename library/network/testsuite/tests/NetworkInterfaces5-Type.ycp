// bnc#704999
{

include "testsuite.ycp";
import "Assert";
import "NetworkInterfaces";

typedef map<string, any> ifcfg_t;

ifcfg_t mybond = $[
    "BOOTPROTO": "dhcp",
    "BONDING_MASTER": "yes",
    "BONDING_SLAVE_0": "eth0",
    "BONDING_SLAVE_1": "eth1",
];

ifcfg_t mybridged = $[
    "BOOTPROTO": "dhcp",
    "BRIDGE": "yes",
    "BRIDGE_PORTS": "eth0 tap0",
];

ifcfg_t myvirt = $[
    "BOOTPROTO": "static",
    "TUNNEL": "tap",
];

DUMP("NetworkInterfaces::GetTypeFromIfcfg");
TEST(``(
        Assert::Equal("bond", NetworkInterfaces::GetTypeFromIfcfg(mybond))
        ), [], nil);
TEST(``(
        Assert::Equal("br", NetworkInterfaces::GetTypeFromIfcfg(mybridged))
        ), [], nil);
TEST(``(
        Assert::Equal("tap", NetworkInterfaces::GetTypeFromIfcfg(myvirt))
        ), [], nil);

/*EOF*/
}
