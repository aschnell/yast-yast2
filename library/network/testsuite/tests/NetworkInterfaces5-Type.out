Dump	NetworkInterfaces::GetTypeFromIfcfg
Return	true
Return	true
Return	true
